### ISSUE TITLE
task 'test:server' failed after upgrade to Kibana 5.6.7

### ISSUE THREAD

**Issue by arlept** [elastic/kibana-plugin-helpers#67]
Hello,

Context: I'm migrating custom Kibana plugins from Kibana 5.2.2 to Kibana 5.6.7.
I updated my `package.json`, upgraded my code, tested with `npm run start` and everything is fine.
I'm using now "@elastic/plugin-helpers": "8.1.3" which seems to be the latest version compatible with my setup.
I did not run the "template-kibana-plugin" again to do this migration.

What's happening: `npm run test:server` fails with following error:

```
Error: Cannot find module 'D:\Users\xxxxxx\src\kibana\test\mocha_setup.js'
```

My understanding: the thing is that `mocha_setup.js` does not belong to Kibana 5.6.7 anymore. It seems that there were plans to move scripts ([kibana#11095](https://github.com/elastic/kibana/issues/11095)) with impacts on makelogs and mocha npm scripts ([kibana #11665](https://github.com/elastic/kibana/pull/11665)) which were backported to 5.x.
Plugin Helpers still reference this `mocha_setup.js` script in this version ([link](https://github.com/elastic/kibana-plugin-helpers/blob/master/tasks/test/server/test_server_action.js#L9): 
```js
const mochaSetupJs = resolve(plugin.kibanaRoot, 'test/mocha_setup.js');
```
I think the `test_server_action.js` file needs a patch.

I know my setup is deprecated, but is there any chance that this gets fixed? Any idea of a workaround?

**Comment by w33ble** [comment]
I'm fairly certain you want to be using 7.x of the plugin helpers. 8.x and 9.x are more targeting some recent changes in Kibana master, and 8.x is basically deprecated already. The compat chart in the readme explicitly only mentiones 7.x and then the embedded helpers for a reason.

Give 7.1.5 a try and see if you are still having issues.

**Comment by w33ble** [comment]
Actually, sorry, I think you're right, there is a bug here. [That script was removed in 5.6](https://github.com/elastic/kibana/tree/5.6/test). 

@spalger @jbudz I recall you two being the ones working on all the script changes. Any insight here, and maybe one of you are open to fixing this in 7.x of the plugin helpers?

**Comment by arlept** [comment]
Thank you for pointing this part of the source, it directed me to the part I was missing to propose a fix: doing the optimize before running tests.

The trick is to force plugin helper to use `mocha.opts` provided by Kibana, then force node to look at both plugin's and Kibana modules (caveat applies here?).

Here is [a gist](https://gist.github.com/arlept/d8dad2db15acf125227806ba500accc9) of rewritten `test_server_action.js` which works fine for me; if you think it can be integrated, I'd be happy.

I guess there is still similar issue in `test:client`, though I don't use them so cannot propose a change.

**Comment by spalger** [comment]
The `test:browser` task should be fine, it uses a pretty different setup and doesn't rely on the mocha_setup script that was removed in https://github.com/elastic/kibana/pull/15281. https://github.com/elastic/kibana-plugin-helpers/pull/68 should fix the issue. If you want to try it out locally I would appreciate it @arlept.

Something like this should do it
```sh
# clone my fork of the repo
git clone --branch 7.x git@github.com:spalger/kibana-plugin-helpers.git
cd kibana-plugin-helpers
yarn # install deps
yarn link # allows you to link this into your plugin
cd ../../your-kibana-plugin
yarn link @elastic/plugin-helpers
```

**Comment by w33ble** [comment]
Closed by #68 

I published 7.1.6 as well, so @arlept you should be all set by just using that version.

**Comment by arlept** [comment]
Hello, thank  you both, 7.1.6 works like a charm !
